src/v3/src/hooks/useInteractionCodeFlow.ts

```diff
@@ -0,0 +1,139 @@
+      return;
+    }
+
+    if (!([IdxStatus.TERMINAL, IdxStatus.SUCCESS].includes(transaction.status)
+        || !transaction.nextStep)) {
+      setFormBag(undefined);
+      return;
+    }
+
+    if (!(isOauth2Enabled(widgetProps) && transaction.interactionCode)) {
+      setFormBag(undefined);
+      return;
+    }
+
+    const {
+      authClient,
+      redirectUri,
+      redirect,
+    } = widgetProps;
+    const { interactionCode } = transaction;
+    const transactionMeta = authClient.idx.getSavedTransactionMeta();
+    const state = authClient.options.state || transactionMeta?.state;
+    const redirectParams = { interaction_code: interactionCode || '', state: state || '' };
+
+    const isRemediationMode = getMode(widgetProps) === 'remediation';
+    if (isRemediationMode) {
+      authClient.idx.clearTransactionMeta();
+    }
+
+    const shouldRedirect = redirect === 'always';
+    if (shouldRedirect) {
+      if (!redirectUri) {
+        throw new Error('redirectUri is required');
+      }
+
+      const searchParams = new URLSearchParams(redirectParams);
+      const redirectFormBag = redirectTransformer(
+        transaction,
+        `${redirectUri}?${searchParams.toString()}`,
+        widgetProps,
+      );
+      setFormBag(redirectFormBag);
+      return;
+    }
+
+    if (isRemediationMode) {
+      onSuccess?.({
+        status: IdxStatus.SUCCESS,
+        ...redirectParams,
+      });
+      setFormBag(undefined);
+      return;
+    }
+
+    // Operating in "relying-party" mode. The widget owns this transaction.
+    // Complete the transaction client-side and call success/resolve promise
+    if (!transactionMeta) {
+      throw new Error('Could not load transaction data from storage');
+    }
+
+    const { codeVerifier } = transactionMeta;
+    const exchangeCodeForTokens = async () => {
+      await authClient.token.exchangeCodeForTokens({ codeVerifier, interactionCode })
+        .then(({ tokens }) => {
+          const result: RenderResult = { tokens, status: IdxStatus.SUCCESS };
+          setFormBag(undefined);
+          onSuccess?.(result);
+        })
+        .catch((error: AuthSdkError) => {
+          const typedError = getTypedOAuthError(error);
+          const shouldDisplayError = typedError instanceof RecoverableError
+            && (typedError.is('terminal') || typedError.is('inline'));
+          if (shouldDisplayError) {
+            const terminalErrorFb = createForm();
+            terminalErrorFb.uischema.elements.push({
+              type: 'InfoBox',
+              options: {
+                message: {
+                  class: 'ERROR',
+                  i18n: { key: '' },
+                  message: typedError.errorDetails.errorSummary
+                    || loc('oform.error.unexpected', 'login'),
+                },
+                class: 'ERROR',
+              },
+            } as InfoboxElement);
+            setFormBag(terminalErrorFb);
+          } else {
```

Comment:
For information as to _why_ we are doing this, you can see this: https://github.com/okta/okta-signin-widget/pull/2734